packages/io-services-cms-models/src/lib/fsm/store.cosmos.ts

```diff
@@ -55,6 +59,53 @@ export const createCosmosStore = <
       )
     );
 
+  const buildReadOperations = (ids: string[]): ReadOperationInput[] =>
+    ids.map((id) => ({
+      partitionKey: id,
+      operationType: BulkOperationType.Read,
+      id,
+    }));
+
+  const bulkFetch = (ids: string[]) =>
+    pipe(
+      // bulk fetch of items by id
+      TE.tryCatch(
+        () =>
+          container.items.bulk(buildReadOperations(ids), {
+            continueOnError: true,
+          }),
+        (err) =>
+          new Error(
+            `Failed to bulk read items from database, ${E.toError(err).message}`
+          )
+      ),
+      TE.map((operationResponses) =>
+        operationResponses.map((res) =>
+          res.statusCode === 404
+            ? // if the item isn't found, it's ok
+              O.none
+            : // if present, try to decode in the expected shape
+              pipe(
+                {
+                  ...res.resourceBody,
+                  last_update: res.resourceBody
+                    ? new Date(
+                        // eslint-disable-next-line no-underscore-dangle
+                        (res.resourceBody._ts as number) * 1000
+                      ).toISOString() // Unix timestamp
+                    : new Date().toISOString(),
+                  version: res.eTag,
+                },
+                codec.decode,
+                E.fold(
+                  () => O.none,
+                  (a) => O.some(a)
+                )
+              )
+        )
+      )
```

Comment:
nitpick: IMHO this way it's more readable 😉
```suggestion
      TE.map((operationResponses) =>
        operationResponses.map(
          flow(
            O.fromPredicate((res) => res.statusCode === 404),
            // if present, try to decode in the expected shape
            O.chain((res) =>
              pipe(
                {
                  ...res.resourceBody,
                  last_update: res.resourceBody
                    ? new Date(
                        // eslint-disable-next-line no-underscore-dangle
                        (res.resourceBody._ts as number) * 1000
                      ).toISOString() // Unix timestamp
                    : new Date().toISOString(),
                  version: res.eTag,
                },
                codec.decode,
                E.fold(() => O.none, O.some)
              )
            )
          )
        )
      )
```